java-runtime/src/main/proto/spine/validate/validation_error.proto

```diff
@@ -65,8 +65,44 @@ message ConstraintViolation {
     // The name of the validated message type.
     string type_name = 7;
 
-    // A path to the field in an invalid value.
-    base.FieldPath field_path = 3;
+    // A field that violates the validation constraint.
+    oneof field {
+
+        // The field path.
+        //
+        // This field is populated when this violation is nested in another constraint violation.
+        // In-depth validation of fields is performed when the `(validate)` option is applied.
+        //
+        // Consider the following example:
+        //
+        // ```
+        // message Student {
+        //     Email email = 1 [(validate) = true];
+        // }
+        // message Email {
+        //     string value = 1 [(pattern).regex = "^[\\w.-]+@[\\w.-]+\\.[a-zA-Z]{2,}$"];
+        // }
+        // ```
+        //
+        // In this case, each time `Student` message is built, its nested `email` message is
+        // also fully validated according to its own constraints. If an invalid `Email` instance
+        // is passed to `Student`, two constraint violations will be created:
+        //
+        // 1. The parental violation, which has `field_name` set to `email`. The error message
+        //    indicates an invalid field, but without any specifics.
+        // 2. The nested violation which has `field_path` set to `email.value`. The error message
+        //    contains more details about what exactly gone wrong.
+        //
+        // The nested violation will be in `violation` list of the parental one.
+        //
+        base.FieldPath field_path = 3;
```

Comment:
One more question here is why we cannot go with just this field. There may be a `FieldPath` with a single "path element" in it.